source/extensions/filters/http/common/jwks_fetcher.cc

```diff
@@ -106,19 +146,75 @@ class JwksFetcherImpl : public JwksFetcher,
   Upstream::ClusterManager& cm_;
   bool complete_{};
   JwksFetcher::JwksReceiver* receiver_{};
-  const envoy::config::core::v3::HttpUri* uri_{};
+
   Http::AsyncClient::Request* request_{};
 
+  Tracing::Span* parent_span_{};
+
+  const envoy::extensions::filters::http::jwt_authn::v3::RemoteJwks& remote_jwks_;
+  const envoy::config::core::v3::HttpUri& uri_;
+
+  Envoy::Event::Dispatcher& dispatcher_;
+
+  Envoy::BackOffStrategyPtr backoff_strategy_;
+
+  uint32_t num_retries_;
+
+  uint32_t retries_remaining_;
+
+  Envoy::Random::RandomGeneratorImpl random_;
+
   void reset() {
     request_ = nullptr;
     receiver_ = nullptr;
-    uri_ = nullptr;
+    parent_span_ = nullptr;
+
+    // truncated backoff back to 0 retries attempted.
+    retries_remaining_ = num_retries_;
+
+    if (backoff_strategy_) {
```

Comment:
no need this if check since we always create it

Reply:
done